flink-dist/src/main/flink-bin/NOTICE

```diff
@@ -407,3 +391,86 @@ LongAdder), which was released with the following comments:
     http://creativecommons.org/publicdomain/zero/1.0/
 
 
+-----------------------------------------------------------------------
+                           Snappy-Java
+-----------------------------------------------------------------------
+
+This product includes software developed by Google
+ Snappy: http://code.google.com/p/snappy/ (New BSD License)
+
+This product includes software developed by Apache
+ PureJavaCrc32C from apache-hadoop-common http://hadoop.apache.org/
+ (Apache 2.0 license)
+
+This library containd statically linked libstdc++. This inclusion is allowed by
+"GCC RUntime Library Exception"
+http://gcc.gnu.org/onlinedocs/libstdc++/manual/license.html
+
+== Contributors ==
+  * Tatu Saloranta
+    * Providing benchmark suite
+  * Alec Wysoker
+    * Performance and memory usage improvement
+
+
+-----------------------------------------------------------------------
+                       Apache Commmons Compress
+-----------------------------------------------------------------------
+
+Apache Commons Compress
+Copyright 2002-2015 The Apache Software Foundation
+
+This product includes software developed at
+The Apache Software Foundation (http://www.apache.org/).
+
+The files in the package org.apache.commons.compress.archivers.sevenz
+were derived from the LZMA SDK, version 9.20 (C/ and CPP/7zip/),
+which has been placed in the public domain:
+
+"LZMA SDK is placed in the public domain." (http://www.7-zip.org/sdk.html)
+
+
+-----------------------------------------------------------------------
```

Comment:
I think that Till is right, we need to include the NOTICE files from the bundled dependencies. The binary NOTICE file is for convenience anyways, but if we provide it, it needs to forward the dependencies' NOTICE files.